### DISCUSSION TITLE
My ideal config schema/layout

### DISCUSSION THREAD

**Discussion by silveltman** [RyanClementsHax/tailwindcss-themer#66] [Ideas]
I would love the config to be as follows:

```
module.exports = {
	plugins: [
		require('tailwinddd-themer'),
	],
	themer: {
		default: {
			// colors
			// etc.
		},
		dark: {
			// colors
			// etc.
		}
		// etc.
	}
};
```

Improvements (in my opinion):
- sepearate config object, instead of in the plugin itself.
- removed 'extend' from themes, I did not see the benefit of this (correct me if I'm wrong)
- from themes as an array to an object with objects, where the object key is the theme name

Let me know what you guys think! :)

**Comment by RyanClementsHax** [comment]
I considered the option when first making the plugin. I'd need to think a little more to remember exactly why but I believe it had something to do with tailwinds plugin api making such a structure difficult for generating css vars. I'll get back to this when I have more info.

**Reply by RyanClementsHax** [reply]
Alright, I remember why I did this. It's because of a limitation in the tailwindcss plugin api. Please upvote https://github.com/tailwindlabs/tailwindcss/discussions/6980 this discussion if you want more progress on this. Btw, I agree with you that I'd like more config specified on the tailwind config object directly. Until then, we have to work with what we have now (I'm totally open to other options btw).

**Reply by silveltman** [reply]
Upvoted ✅ 

And no worries, already my favourite tailwind plugin!

**Reply by RyanClementsHax** [reply]
aw thanks <3
